src/Microsoft.Health.Fhir.SqlServer/Features/Search/HashingSqlQueryParameterManager.cs

```diff
@@ -195,7 +195,9 @@ private static void WriteAndAdvance<T>(Span<byte> buffer, ref int currentIndex,
                 Debug.Assert(buffer.Length >= elementLength, "Initial buffer size is not large enough for the datatypes we are trying to write to it");
             }
 
+#pragma warning disable CS9191 // The 'ref' modifier for an argument corresponding to 'in' parameter is equivalent to 'in'. Consider using 'in' instead. Justification = Required for .NET6 compat.
```

Comment:
Suggested alternate below - unsure which is better but this is another potential way. 😄 

``` csharp
#if NET8_0_OR_GREATER
            MemoryMarshal.Write(buffer[currentIndex..], in element);
#else
            MemoryMarshal.Write(buffer[currentIndex..], ref element);
#endif

Reply:
Yeah I don't know, since in the description it said they were _equivalent_ I wasn't sure it mattered?

Reply:
I personally prefer the `#if NET8_0_OR_GREATER` because it's more obvious in the future if we deprecate .NET6. But this is splitting hairs - both do the same. 🚀